File: markdown/src/main/java/ru/noties/markwon/extension/emoji/Emoji.java
package ru.noties.markwon.extension.emoji;

import org.commonmark.node.CustomNode;
import org.commonmark.node.Delimited;

/**
 * Created by kosh on 20/08/2017.
 */

public class Emoji extends CustomNode implements Delimited {

    private static final String DELIMITER = ":";
    private String emoji;

    @Override public String getOpeningDelimiter() {
        return DELIMITER;
    }

    @Override public String getClosingDelimiter() {
        return DELIMITER;
    }

    public String getEmoji() {
        return emoji;
    }

    public void setEmoji(String emoji) {
        this.emoji = emoji;
    }
}

File: app/src/main/java/com/fastaccess/github/platform/markwon/emoji/Emoji.kt
package com.fastaccess.github.platform.markwon.emoji

import org.commonmark.node.CustomNode
import org.commonmark.node.Delimited

/**
 * Created by kosh on 20/08/2017.
 */

class Emoji : CustomNode(), Delimited {
    var emoji: String? = null

    override fun getOpeningDelimiter(): String = DELIMITER

    override fun getClosingDelimiter(): String = DELIMITER

    companion object {
        private const val DELIMITER = ":"
    }
}
